autotest: restore netcdf_cf checks

dbaston opened the pull request:
<!--
IMPORTANT: Do NOT use GitHub to post any questions or support requests!
           They will be closed immediately and ignored.

Make sure that the title of your commit(s) is descriptive. Typically, they
should be formatted as "component/filename: Describe what the commit does (fixes #ticket)",
so that anyone that parses 'git log' immediately knows what a commit is about.
Do not hesitate to provide more context in the longer part of the commit message.

GOOD: "GTiff: fix wrong color interpretation with -co ALPHA=YES (fixes #1234)

When -co ALPHA=YES was used, but PHOTOMETRIC was not specified, the ExtraSample
tag was wrongly set to unspecified.
"

BAD: "Fix crash", "fix #1234"

In case you need several iterations to make continuous integration happy,
please squash your commits in a single one at the end. See
[Contributing](https://github.com/OSGeo/gdal/blob/master/CONTRIBUTING.md)
-->

## What does this PR do?

Updates netCDF CF-compliance tests to:

- Remove use of discontinued HTTP checking service
- Remove bundled version of cf-checker
- Avoid manual download of XML files
- Make tests skip if cf checker is not available instead of silently succeeding
- Make tests fail if files to be checked do not exist
- Remove error codes and logging in favor of assert statements

## What are related issues/pull requests?

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

rouault commented:
(the fedoraw_rawhide failure should be fixed by rebasing on top of latest master)

dbaston commented:
It looks like 18.04 has a problem installing the netCDF python package on which `cfchecker` depends.

Maybe rather than adding `cfchecker` to `requirements.txt` I should manually install it in one of the newer CI environments?

`docker run -it --rm ghcr.io/osgeo/gdal-deps:ubuntu_18.04-master sh -c "pip3 install -U setuptools && pip3 install netCDF4"`

```
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/b0/3a/88b210db68e56854d0bcf4b38e165e03be377e13907746f825790f3df5bf/setuptools-59.6.0-py3-none-any.whl (952kB)
    100% |████████████████████████████████| 962kB 1.3MB/s 
Installing collected packages: setuptools
  Found existing installation: setuptools 39.0.1
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
Successfully installed setuptools-59.6.0
Collecting netCDF4
  Downloading https://files.pythonhosted.org/packages/a6/f9/12f43c2e31d1c7aae0827820ae747fa4c7cc382b38dab11e8c11575d3af6/netCDF4-1.6.2.tar.gz (777kB)
    100% |████████████████████████████████| 778kB 1.7MB/s 
Requirement already satisfied: numpy>=1.9 in /usr/lib/python3/dist-packages (from netCDF4)
Collecting cftime (from netCDF4)
  Downloading https://files.pythonhosted.org/packages/20/3f/607a69f1253e5c8835400641da91a6b54cc49e3e9f6ac5848a745638f9ea/cftime-1.6.0.tar.gz (47kB)
    100% |████████████████████████████████| 51kB 4.7MB/s 
Building wheels for collected packages: netCDF4, cftime
  Running setup.py bdist_wheel for netCDF4 ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ap_cm9mr/netCDF4/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp1jl3pyk2pip-wheel- --python-tag cp36:
  reading from setup.cfg...
  using nc-config...
  checking /usr/include ...
  hdf5 headers not found in /usr/include
  checking /usr/include/hdf5/serial ...
  HDF5 library version: 1.10.0-patch1 headers found in /usr/include/hdf5/serial
  using netcdf library version b'4.6.0'
  using Cython to compile netCDF4.pyx...
  netcdf lib has group rename capability
  netcdf lib has nc_inq_path function
  netcdf lib has nc_inq_format_extended function
  netcdf lib has nc_open_mem function
  netcdf lib does not have nc_create_mem function
  netcdf lib does not have cdf-5 format capability
  netcdf lib does not have netcdf4 parallel functions
  netcdf lib does not have pnetcdf parallel functions
  netcdf lib does not have bit-grooming/quantization functions
  netcdf lib does not have zstandard compression functions
  netcdf lib does not have bzip2 compression functions
  netcdf lib does not have blosc compression functions
  netcdf lib does not have szip compression functions
  netcdf lib does not have nc_set_alignment function
  netcdf lib does not have nc_inq_filter_avail function
  NETCDF_PLUGIN_DIR not set, no netcdf compression plugins installed
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/netCDF4
  copying src/netCDF4/utils.py -> build/lib.linux-x86_64-3.6/netCDF4
  copying src/netCDF4/__init__.py -> build/lib.linux-x86_64-3.6/netCDF4
  package init file 'src/netCDF4/plugins/__init__.py' not found (or not a regular file)
  running build_ext
  building 'netCDF4._netCDF4' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/src
  creating build/temp.linux-x86_64-3.6/src/netCDF4
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/usr/include -I/usr/include/hdf5/serial -I/usr/lib/python3/dist-packages/numpy/core/include -Iinclude -I/usr/include/python3.6m -c src/netCDF4/_netCDF4.c -o build/temp.linux-x86_64-3.6/src/netCDF4/_netCDF4.o
  x86_64-linux-gnu-gcc: error: src/netCDF4/_netCDF4.c: No such file or directory
  x86_64-linux-gnu-gcc: fatal error: no input files
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for netCDF4
  Running setup.py clean for netCDF4
  Running setup.py bdist_wheel for cftime ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ap_cm9mr/cftime/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpbe8bi0ampip-wheel- --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/cftime
  copying src/cftime/__init__.py -> build/lib.linux-x86_64-3.6/cftime
  running build_ext
  building 'cftime._cftime' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/tmp
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime/src
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime/src/cftime
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/python3.6m -c /tmp/pip-build-ap_cm9mr/cftime/src/cftime/_cftime.c -o build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime/src/cftime/_cftime.o
  x86_64-linux-gnu-gcc: error: /tmp/pip-build-ap_cm9mr/cftime/src/cftime/_cftime.c: No such file or directory
  x86_64-linux-gnu-gcc: fatal error: no input files
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for cftime
  Running setup.py clean for cftime
Failed to build netCDF4 cftime
Installing collected packages: cftime, netCDF4
  Running setup.py install for cftime ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ap_cm9mr/cftime/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-3rwnjt7q-record/install-record.txt --single-version-externally-managed --compile:
    running install
    /usr/local/lib/python3.6/dist-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      setuptools.SetuptoolsDeprecationWarning,
    running build
    running build_py
    running build_ext
    building 'cftime._cftime' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/tmp
    creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr
    creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime
    creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime/src
    creating build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime/src/cftime
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/python3.6m -c /tmp/pip-build-ap_cm9mr/cftime/src/cftime/_cftime.c -o build/temp.linux-x86_64-3.6/tmp/pip-build-ap_cm9mr/cftime/src/cftime/_cftime.o
    x86_64-linux-gnu-gcc: error: /tmp/pip-build-ap_cm9mr/cftime/src/cftime/_cftime.c: No such file or directory
    x86_64-linux-gnu-gcc: fatal error: no input files
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ap_cm9mr/cftime/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-3rwnjt7q-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-ap_cm9mr/cftime/
```

rouault commented on .github/workflows/alpine/Dockerfile.ci:
```diff
@@ -25,3 +25,4 @@ RUN apk add \
 
 COPY requirements.txt /tmp/
 RUN python3 -m pip install --break-system-packages -U -r /tmp/requirements.txt
+RUN python3 -m pip install cfchecker
```
```suggestion
RUN python3 -m pip install --break-system-packages cfchecker
```

rouault commented:
I suspect the fedora_rawhide failure is unrelated to this PR. For some unknown reason, the ghcr.io/osgeo/gdal-deps:fedorarawhide-master image is somehow corrupted with a broken libssl.so.3 symlink

dbaston commented:
I removed the `cfchecker` package addition from alpina and fedora (there seems to be a problem installing it with Python 3.12) and added it to Ubuntu 20.04.  Upon further reflection it is probably a good idea to minimize package installs on the rolling distributions.
